workflow/controller/workflowpod.go

```diff
@@ -332,12 +333,27 @@ func (woc *wfOperationCtx) createWorkflowPod(nodeName string, mainCtr apiv1.Cont
 			return nil, errors.Wrap(err, "", "Error in Unmarshalling after merge the patch")
 		}
 	}
+
+	// we must check to see if the pod exists rather than just optimistically creating the pod and see if we get
+	// an `AlreadyExists` error because we won't get that error if there is not enough resources
+	obj, exists, err := woc.controller.podInformer.GetStore().Get(cache.ExplicitKey(pod.Namespace + "/" + pod.Name))
```

Comment:
This solution depends on the informer being correct when queries here. 

If the informer is incorrect, i.e believe that a pod has not been created when it in fact has, then the workflow will error.

An alternative solution is to do `podInterface.Get(pod.Name)` and see what error it returns (i.e. is it an `apierr.IsNotFound(..)`?).